src/com/owncloud/android/ui/activity/FileDisplayActivity.java

```diff
@@ -1632,7 +1632,7 @@ private void onCreateFolderOperationFinish(CreateFolderOperation operation,
         } else {
             try {
                 showSnackMessage(
-                    ErrorMessageAdapter.getResultMessage(result, operation, getResources())
+                        FileStorageUtils.getNewFolderName(((CreateFolderOperation) operation).getRemotePath()) + " " + getResources().getString(R.string.folder_already_exists)
```

Comment:
Can't you get the folder name from https://github.com/sd1998/android/blob/master/src/com/owncloud/android/ui/errorhandling/ErrorMessageAdapter.java#L106 by calling `operation.getFileName()` and avoid creating a new method in FileStorageUtils?

Reply:
@davigonz The getFileName() method is applicable to UploadFileOperation.class and not for the CreateFolderOperation.class thus cannot use it.

Reply:
I see...but since what you are doing is generating an error message, don't you think it would be better to add a new if case in https://github.com/sd1998/android/blob/master/src/com/owncloud/android/ui/errorhandling/ErrorMessageAdapter.java#L105  - `if (operation instanceof CreateFolderOperation)` and create a new method in `CreateFolderOperation` with something like:

```java
public String getFolderToCreateName() {
    return new OCFile(mRemotePath).getFileName();
}
```

Reply:
@davigonz I just changed the getRemotePath() function in CreateFolderOperation.java to getNewFolderName() which would do exactly as you suggested in the getFolderToCreateName() function in previous [comment](https://github.com/owncloud/android/pull/2248#discussion_r202600158). Hence not requiring the getNewFolderName() in FileStorageUtils.java hence string manipulation.